### PR TITLE
Added dependency inversion principle in SOLID 

### PR DESCRIPTION
**Pull request by luismateoh**
- Added dependecy inversion principle
- Updated the reference in SOLID list

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -644,6 +644,34 @@ Resources
...
+relationships with high-level abstractions. This means that classes should not depend on low-level details, emphasizing
+the use of interfaces and abstract classes.
+
+Why
+
+- By decoupling the high-level modules from the low-level modules, the high-level modules become more reusable and
+  maintainable.
+- Facilitates unit testing by allowing the use of mock objects, enabling isolated testing of modules.
+- Reduces the risk of breaking the system when changes are made.
+- Allows to add new implementations without changing the existing code, enhancing the system's extensibility.
```
Comment (nchinling) [inline README.md]: Consider 'Allows adding new...'

File: README.md [hunk 2]
```diff
@@ -670,8 +670,8 @@ How
 Resources
 
 - [SOLID — Dependency Inversion Principle](https://medium.com/@inzuael/solid-dependency-inversion-principle-part-5-f5bec43ab22e)
-- [Dependency Injection (wikipedia.org)](https://en.wikipedia.org/wiki/Dependency_injection)
-
+- [DDependency inversion principle (wikipedia.org)](https://en.wikipedia.org/wiki/Dependency_inversion_principle)
```
Comment (nchinling) [inline README.md]: Remove the additional letter 'D' in DDependency

File: README.md [hunk 3]
```diff
@@ -646,11 +646,10 @@ Resources
...
 ## Dependency Inversion Principle
 
-Dependency Injection is an Inversion of Control technique for supplying objects (“dependencies”) to a class by a way of
-the Dependency Injection Design Pattern. Typically passing dependencies via constructor or field. In contrast the
-Dependency Inversion Principle is a general design guideline which recommends that classes should only have direct
-relationships with high-level abstractions. This means that classes should not depend on low-level details, emphasizing
-the use of interfaces and abstract classes.
+Dependency Inversion is the strategy of depending upon interfaces or abstract functions and classes rather than upon
+concrete functions and classes.
+Simply put, when components of our system have dependencies, we don’t want directly inject a component’s dependency into
```
Comment (nchinling) [inline README.md]: Missing a 'to' in 'we don’t want directly'

**Comment by webpro** [comment]
Thanks @luismateoh!

- Any chance you could work @nchinling's suggestions in?
- Can you please add a link to the section in the content overview at the top?
- Would be great if you could limit line width to 80 like the rest of the document.

**Comment by luismateoh** [comment]
I made the corrections, thanks @nchinling for the feedback.
I added the link on the content overview and formated the section with 80 line limit.

**Comment by webpro** [comment]
Thanks @luismateoh!

**Pull request merged**